integration-tests/search.test.ts

```diff
@@ -513,3 +514,242 @@ describe('search', () => {
+            client,
+            { url: 'Patient', params: { _include: 'Patient:organization' } },
+            {
+                resourceType: 'Organization',
+            },
+        );
+
+        await expectResourceToNotBePartOfSearchResults(
+            client,
+            { url: 'Patient', params: { _revinclude: 'Encounter:subject' } },
+            encounterResource,
+        );
+
+        // failing searches (no scopes for Observation, so should reject)
+        await expect(client.get('Observation')).rejects.toMatchObject({
+            response: { status: 401 },
+        });
+
+        await expect(client.get('Observation', { params: { _include: 'Observation:subject' } })).rejects.toMatchObject({
+            response: { status: 401 },
+        });
+    });
+
+    test('tests with additional patient scopes', async () => {
+        const orgClient = await getFhirClient(
+            'fhirUser patient/Patient.read patient/Organization.read launch/patient profile openid',
+            false,
+        );
+        const encClient = await getFhirClient(
+            'fhirUser patient/Patient.read patient/Encounter.read launch/patient profile openid',
+            false,
+        );
+
+        await expectResourceToNotBePartOfSearchResults(
+            orgClient,
+            { url: 'Patient', params: { _include: 'Patient:organization' } },
+            {
+                resourceType: 'Organization',
+            },
+        );
+
+        await expectResourceToBePartOfSearchResults(
+            encClient,
+            { url: 'Patient', params: { _revinclude: 'Encounter:subject' } },
+            encounterResource,
+        );
+    });
+});
+
+// these are largely similar to above, but have different behavior and so are kept separate
+describe('searches with user permission levels', () => {
+    let adminClient: AxiosInstance;
+    let encounterResource: any;
+
+    beforeAll(async () => {
+        adminClient = await getFhirClient('fhirUser user/*.*', true);
+        const encounterResourceJson = {
+            resourceType: 'Encounter',
+            status: 'in-progress',
+            class: {
+                system: 'http://terminology.hl7.org/CodeSystem/v3-ActCode',
+                code: 'IMP',
+                display: 'inpatient encounter',
+            },
+            subject: {
+                reference: `Patient/92e0d921-bb19-4cae-a3cc-9d3c5bcf7a39`,
+                display: 'Sherlock Holmes',
+            },
+        };
+        encounterResource = (await adminClient.post('Encounter', encounterResourceJson)).data;
+        await waitForResourceToBeSearchable(adminClient, encounterResource);
+    });
+
+    afterAll(async () => {
+        await adminClient.delete(`Encounter/${encounterResource.id}`);
+    });
+
+    test('tests with only patient.read scope', async () => {
+        const clientAsPatient = await getFhirClient('fhirUser user/Patient.read profile openid', false);
```

Comment:
I would make this its own `describe` or you can do the `case` example I gave above